### ISSUE TITLE
existing resource conflict

### ISSUE THREAD

**Issue by tarunnarang0201** [helm/helm#8350]
There are secrets which should not be deleted in helm and for that annotation is applied ("helm.sh/resource-policy": keep). 

**Secrets.yaml**
```
apiVersion: v1
kind: Secret
metadata:
  name: resources-secret
  namespace: resources
  annotations:
    "helm.sh/resource-policy": keep

```
Even after adding the annotation, secrets are deleted when release is deleted/uninstalled. 
Having investigated further, it was found, if the namespace is deleted as a part of the release, it would also delete the secrets no matter whether or not annotation ("helm.sh/resource-policy": keep) exists.

**Namespace.Yaml file**

```
apiVersion: v1
kind: Namespace
metadata:
  name: resources

```

Now to ensure namespace also does not delete when the release is deleted. Annotation ("helm.sh/resource-policy": keep) is added in namespace too and the updated file looks like this:-

**Updated Namespace.yaml**

```
apiVersion: v1
kind: Namespace
metadata:
  name: resources
  annotations:
    "helm.sh/resource-policy": keep

```
After applying these changes, **when the release is deleted it works what was expected** and it did not delete secrets and namespace but other resources. 

However, when the release is redeployed, **the below error is received** :-

command executed - helm upgrade --install resources-uat resources
```
Release "resources-uat" does not exist. Installing it now.
Error: rendered manifests contain a resource that already exists. Unable to continue with install: existing resource conflict: namespace: , name: resources, existing_kind: /v1, Kind=Namespace, new_kind: /v1, Kind=Namespace

```
Here, **expected result** was, helm should have skipped the namespace and secrets resources and deployed rest of the resources.

**Comment by bacongobbler** [comment]
> Having investigated further, it was found, if the namespace is deleted as a part of the release, it would also delete the secrets no matter whether or not annotation ("helm.sh/resource-policy": keep) exists.

This makes sense. You asked Helm to manage the creation and deletion of the namespace. When you ran `helm uninstall`, the default deletion policy was Helm told Kubernetes to delete the namespace. When a namespace is deleted, all objects residing in that namespace are deleted. Helm's doing the right thing here based on the instructions provided by the user.

It's unclear to me what you are suggesting we should fix here.

> However, when the release is redeployed, **the below error is received**

Can you please share what version of Helm you are using? This should be fixed if you try with a newer version of Helm in a fresh cluster.

**Comment by tarunnarang0201** [comment]
`It's unclear to me what you are suggesting we should fix here.`
Here I am not expecting anything, I meant the same thing which you mentioned above.

I was expecting to fix an error, in case annotation is added to not delete a namespace and while re deploying, it should not have thrown an error.    

I am using version 3.1.2, Has it been fixed in 3.2.3

**Comment by bacongobbler** [comment]
Yes.
